library/net_sockets.c

```diff
@@ -433,6 +433,58 @@ int mbedtls_net_set_nonblock( mbedtls_net_context *ctx )
 #endif
 }
 
+/*
+ * Check if data is available on the socket
+ */
+
```

Comment:
@sbutcher-arm This hasn't been investigated so far. Is it a question out of mere interest or do you see it blocking the PR?

Reply:
Sadly, this is a blocking issue for me. I need to know either how it can be implemented in Mbed OS, or why I shouldn't care. Without knowing that, I can't approve it.

Reply:
@sbutcher-arm Event-driven I/O is only possible if there is a means to idle a socket until a condition like availability of incoming data, or availability to send outgoing data, is met. The new function shows how this can be implemented on top of Unix or Windows sockets. It has no impact on the usability of Mbed TLS with Mbed OS: Both before and after this PR, use of event-driven I/O in Mbed OS requires some means of polling on the underlying socket.

Reply:
Okay - that satisfies me. Thanks for the explanation @hanno-arm.